include/ttmlir/Dialect/TTIR/IR/TTIRBase.td

```diff
@@ -38,6 +39,6 @@ def TTIR_Dialect : Dialect {
 //===----------------------------------------------------------------------===//
 
 class TTIR_Op<string mnemonic, list<Trait> traits = []> :
-        Op<TTIR_Dialect, mnemonic, traits>;
+        Op<TTIR_Dialect, mnemonic, !listconcat(traits, [Pure])>;
```

Comment:
Should `Pure` trait be the top-level trait? I'm thinking about `alloc` and `dealloc`. Every `dealloc` will trivially be removed since it doesn't produce the result.

Reply:
Good point, we probably should adopt `MemAlloc` and `MemFree` traits for those respectively.